Gems/Terrain/Code/Source/TerrainRenderer/TerrainMeshManager.cpp

```diff
@@ -552,6 +558,11 @@ namespace Terrain
         desc.m_elementFormat = format;
         desc.m_bufferData = initialData;
 
+        if (name != nullptr)
+        {
+            desc.m_bufferName = name;
+        }
+
```

Comment:
I see it has a default `const char* name = ""`. It is still a valid pointer, not nullptr. You may want to change the if check or default the string to nullptr instead.

Reply:
nice catch